setup.py

```diff
@@ -2,7 +2,23 @@
 from setuptools import setup
 from setuptools import Command
 
-__version__ = version = VERSION = '0.1'
+
+def _get_version():
+    """Returns the PROTEUS software version
+
+       Returns
+       -------
+       version : str
+            PROTEUS software version
+    """
+
+    version_file = 'src/proteus/version.py'
+    version = open(version_file).read().split('=')[1]
+    version = version.replace("'", '').replace('"', '').strip()
+    
+    return version
```

Comment:
This feels very trusting that the file is using the correct format, correct style of quote marks, etc.. Could this code be a little more defensive, import `re`, and add a few more robustness checks? Something like:

```
    with open(version_file, 'r') as f:
        text = f.read()

    # Get first match of the version number contained in the version file
    # This regex should match a pattern like: VERSION = '3.2.5', but it
    # allows for varying spaces, number of major/minor versions,
    # and quotation mark styles.
    p = re.search("VERSION[ ]*=[ ]*['\"]\d+([.]\d+)*['\"]", text)

    # Check that the version file contains properly formatted text string
    if p is None:
        raise ValueError(f'Version file {version_file} not properly formatted.'
                     " It should contain text matching e.g. VERSION = '2.3.4'")

    # Extract just the numeric version number from the string
    p = re.search("\d+([.]\d+)*", p.group(0))

    return p.group(0)
```

Reply:
Wow! Thank you, @nemo794 ! Just updated `_get_version()` with your suggestion.